apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -102,6 +102,38 @@ export class AppRoomBridge extends RoomBridge {
 		return this.orch.getConverters()?.get('users').convertById(room.u._id);
 	}
 
+	protected async getMessages(
+		roomId: string,
+		options: {
+			limit: number;
+			skip?: number;
+			sort?: Record<string, 1 | -1>;
+		},
+		appId: string,
+	): Promise<IMessage[]> {
+		this.orch.debugLog(`The App ${appId} is getting the messages of the room: "${roomId}"`);
+		if (typeof options.limit !== 'number') {
+			options.limit = 100;
+		}
```

Comment:
You'd still have to check if limit is NaN since `typeof NaN == 'number'`, `Math.min(NaN, 100)` would still return NaN.

```suggestion
		if (typeof options.limit !== 'number' || Number.isNaN(options.limit)) {
			options.limit = 100;
		}
```

Reply:
Hm actually, there is the special case of 0 as well. Math.min(0, 100) means 0 means all documents again. Negative limit shouldn't be a big deal but might want to not allow that. AFAIK it's driver dependent.

Reply:
```suggestion
		if (typeof options.limit !== 'number' || Number.isNaN(options.limit) || options.limit <= 0) {
			options.limit = 100;
		}
```

Reply:
Let's also make sure we carry this over to the unread messages pr 🙏🏼

Reply:
-Infinity < 0 is true, but I didn't know isFinite is a thing. Thanks.

Reply:
-Infinity < 0 is true. But didn't know isFinite is a thing. Thanks.